plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergBucketing.java

```diff
@@ -0,0 +1,174 @@
+import static io.trino.spi.type.TypeUtils.writeNativeValue;
+import static java.lang.String.format;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+
+public class TestIcebergBucketing
+{
+    private static final TypeManager TYPE_MANAGER = new TestingTypeManager();
+
+    @Test
+    public void testBucketNumberCompare()
+    {
+        // Only primitive types are covered here (logical types like timestamp, date or time maps to long)
+        assertBucketEquals("string", "Trino rocks");
+
+        assertBucketEquals("int", null);
+        assertBucketEquals("int", 300_000);
+        assertBucketEquals("int", Integer.MIN_VALUE);
+        assertBucketEquals("int", Integer.MAX_VALUE);
+
+        assertBucketEquals("long", null);
+        assertBucketEquals("long", 300_000_000_000L);
+        assertBucketEquals("long", Long.MIN_VALUE);
+        assertBucketEquals("long", Long.MAX_VALUE);
+
+        assertBucketEquals("string", null);
+        assertBucketEquals("string", "");
+        assertBucketEquals("string", "test string");
+        assertBucketEquals("string", "\u5f3a\u5927\u7684Trino\u5f15\u64ce"); // 3-byte UTF-8 sequences (in Basic Plane, i.e. Plane 0)
+        assertBucketEquals("string", "\uD843\uDFFC\uD843\uDFFD\uD843\uDFFE\uD843\uDFFF"); // 4 code points: 20FFC - 20FFF. 4-byte UTF-8 sequences in Supplementary Plane 2
+
+        assertBucketEquals("decimal(1, 1)", null);
+        assertBucketEquals("decimal(1, 1)", new BigDecimal(1));
+        assertBucketEquals("decimal(1, 1)", new BigDecimal(1));
+        assertBucketEquals("decimal(18, 10)", new BigDecimal(0));
+        assertBucketEquals("decimal(38, 10)", new BigDecimal("999999.9999999999"));
+        assertBucketEquals("decimal(38, 0)", new BigDecimal("99999999999999999999999999999999999999"));
+        assertBucketEquals("decimal(38, 10)", new BigDecimal("9999999999999999999999999999.9999999999"));
+
+        assertBucketEquals("binary", null);
+        assertBucketEquals("binary", ByteBuffer.wrap(new byte[]{}));
+        assertBucketEquals("binary", ByteBuffer.wrap("hello trino".getBytes(StandardCharsets.UTF_8)));
+        assertBucketEquals("binary", ByteBuffer.wrap("\uD843\uDFFC\uD843\uDFFD\uD843\uDFFE\uD843\uDFFF".getBytes(StandardCharsets.UTF_16)));
+    }
+
+    @Test(dataProvider = "unsupportedBucketingTypes")
+    public void testUnsupportedTypes(Type type)
+    {
+        assertThatThrownBy(() -> computeIcebergBucket(type, null, 1))
+                .hasMessage(format("Cannot bucket by type: %s", type));
+
+        assertThatThrownBy(() -> computeTrinoBucket(type, null, 1))
+                .hasMessage(format("Unsupported type for 'bucket': %s", toTrinoType(type, TYPE_MANAGER)));
+    }
+
+    @Test(dataProvider = "unsupportedTrinoBucketingTypes")
+    public void testUnsupportedTrinoTypes(Type type)
```

Comment:
Iceberg can bucket UUIDs. It appears Trino UUID is not mapped to Iceberg UUIDs - should there be an issue + TODO?

Reply:
I believe there is TODO in the code for UUID handling

Reply:
can you make this test fail when iceberg UUID is mapped to something else than it is today?
so that we don't forget to revisit?
("automated reminder")